packages/firestore/src/core/bundle.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Query } from './query';
+import { SnapshotVersion } from './snapshot_version';
+import { DbBundle, DbNamedQuery } from '../local/indexeddb_schema';
+import * as bundleProto from '../protos/firestore_bundle_proto';
+import { LocalSerializer } from '../local/local_serializer';
+
+/**
+ * Represents a Firestore bundle saved by the SDK in its local storage.
+ */
+export interface Bundle {
+  readonly id: string;
+  readonly version: number;
+  // When the saved bundle is built from the server SDKs.
+  readonly createTime: SnapshotVersion;
+}
+
+/**
+ * Represents a Query saved by the SDK in its local storage.
+ */
+export interface NamedQuery {
+  readonly name: string;
+  readonly query: Query;
+  // When the results for this query are read to the saved bundle.
+  readonly readTime: SnapshotVersion;
+}
+
+/** Encodes a DbBundle to a Bundle. */
+export function fromDbBundle(
+  serializer: LocalSerializer,
+  dbBundle: DbBundle
+): Bundle {
+  return {
+    id: dbBundle.bundleId,
+    createTime: serializer.fromDbTimestamp(dbBundle.createTime),
+    version: dbBundle.version
```

Comment:
What is our versioning strategy? I wonder if we should reject any bundle with a version greater than 1. If so, please file a bug.